### PR TITLE
i18n(ru): Add new guides/typescript translation

### PR DESCRIPTION
**Pull request by sobresaliente**
#### What kind of changes does this PR include?
- Translated content(https://docs.astro.build/en/guides/typescript/)

#### Description
- Translate guides/typescript page to Russian

### CODE REVIEW

File: src/content/docs/ru/guides/typescript.mdx [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+---
+
+<Tag {...props} />
+```
+
+### Определение типов `getStaticPaths()`
+
+<Since v="2.1.0" />
+
+Astro имеет хелперы для работы с типами, которые возвращаются из  [`getStaticPaths()`](/ru/reference/api-reference/#getstaticpaths) функции, которая используется для динамически путей.
```
Comment (RolginRoman) [inline src/content/docs/ru/guides/typescript.mdx:252]: typo
```suggestion
Astro имеет хелперы для работы с типами, которые возвращаются из  [`getStaticPaths()`](/ru/reference/api-reference/#getstaticpaths) функции, которая используется для динамических путей.
```

File: src/content/docs/ru/guides/typescript.mdx [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+i18nReady: true
+---
+
+import Since from '~/components/Since.astro';
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro';
+import Tabs from '~/components/tabs/Tabs';
+
+Astro по умолчанию поддерживает [TypeScript](https://www.typescriptlang.org/). Вы можете импортировать `.ts` и `.tsx` файлы в свой проект, писать Typescript код прямо внутри своих [Astro компонентов](/ru/core-concepts/astro-components/#the-component-script), и даже использовать [`astro.config.ts`](/ru/guides/configuring-astro/#the-astro-config-file) файл, если это необходимо.
+
+Используя Typescript, вы можете предотвратить ошибки во время выполнения описывая объекты и компоненты в коде. К примеру, если вы используете  Typescript для [типизации параметров(пропсов) компонента](#параметры-компонентов), то редактор выведет ошибку, если вы попытаетесь передать параметр, который компонент не принимает.
```
Comment (RolginRoman) [inline src/content/docs/ru/guides/typescript.mdx:13]:
```suggestion
Используя Typescript, вы можете предотвратить ошибки во время выполнения, описывая объекты и компоненты в коде. К примеру, если вы используете  Typescript для [типизации параметров(пропсов) компонента](#параметры-компонентов), то редактор выведет ошибку, если вы попытаетесь передать параметр, который компонент не принимает.
```

Comment (RolginRoman) [reply src/content/docs/ru/guides/typescript.mdx:13]: imho we're losing a bit of the meaning here
en: `...by defining the shapes of objects...`
ru suggestion: ", описывая типы объектов и компонентов"

File: src/content/docs/ru/guides/typescript.mdx [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+
+```ts
+/// <reference types="astro/astro-jsx" />
+
+type MyAttributes = astroHTML.JSX.ImgHTMLAttributes;
+```
+
+:::
+
+### Полиморфичные типы
```
Comment (RolginRoman) [inline src/content/docs/ru/guides/typescript.mdx:228]: not sure about ru alternative here. might it be a "Полиморфные"? 

[wiki Polymorphism (computer science)](https://ru.wikipedia.org/wiki/%D0%9F%D0%BE%D0%BB%D0%B8%D0%BC%D0%BE%D1%80%D1%84%D0%B8%D0%B7%D0%BC_(%D0%B8%D0%BD%D1%84%D0%BE%D1%80%D0%BC%D0%B0%D1%82%D0%B8%D0%BA%D0%B0)) sticks to `Полиморфные` too

```suggestion
### Полиморфные типы
```

**Review by RolginRoman — CHANGES_REQUESTED** [review]

File: src/content/docs/ru/guides/typescript.mdx (hunk 3 above)
Comment (sobresaliente) [reply src/content/docs/ru/guides/typescript.mdx]: Yeah! I agree with you

File: src/content/docs/ru/guides/typescript.mdx (hunk 2 above)
Comment (sobresaliente) [reply src/content/docs/ru/guides/typescript.mdx]: fixed

File: src/content/docs/ru/guides/typescript.mdx (hunk 1 above)
Comment (sobresaliente) [reply src/content/docs/ru/guides/typescript.mdx]: fixed

**Review by RolginRoman — APPROVED** [review]
LGTM 👍🏻

**Comment by Yan-Thomas** [comment]
Thanks @sobresaliente! I can see there's a broken link in `markdown-content.mdx`, would you be up to updating this page here as well and fix the broken link? 🙌

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks peeps, and welcome @sobresaliente! 🥳

**Pull request merged**